sushichef.py

```diff
@@ -72,11 +72,15 @@ def clean_video_title(title, lang_obj):
     return clean_title
 
 
+def generate_source_id(lang_code):
+    source_id = "arvind-{0}-{1}".format(lang_code, uuid.uuid4().hex[:16].lower())
+    return source_id
```

Comment:
This is not a good long term solution since `uuid.uuid4()` will be different every run.
This means topics will have different `source_id` (and by extension different `content_id`  and `node_id`).

See https://ricecooker.readthedocs.io/en/latest/developer/ids.html#node-ids for more info about node ids are generated.

Potential problems:
 - each time we update the channel it will show up to users as "everything has changed" update
 - any URL that includes the node_id from one version of the channel would not work  on the next since node_ids will have changed each run

I would recommend some other change in logic (e.g. using enumerate to get a counter) so it is different for each node, but doesn't change unless website changes. (Note source_id only needs to be unique within parent topic not globally within channel).

Reply:
I added a global source id variable counter @ivanistheone to have the same result, when re-running the script. Thanks for your comments.

Reply:
@mrpau-richard This will only give the same results if the chef has the exact same number of videos for each run. Once videos are added or removed, the source IDs will actually reference other topics in the tree.

Is there a way to get an ID related to the topic itself, e.g. a playlist or channel ID, in the data returned from YouTube?

Reply:
> Is there a way to get an ID related to the topic itself, e.g. a playlist or channel ID, in the data returned from YouTube?

@kollivier there is a channel id returned by the YouTube, but all the videos has the same channel id. 

Please validate my second approach, What I did is for each topics I created a counter and use it with the topic language name as the source id.

Reply:
@mrpau-richard The problem with using a counter is that it is sensitive to changes in sort order. So, let's say the channel author changes the order of the topics for whatever reason. There will still be a topic with the topic ID listed as topic-1, but it may be a different topic from the previous channel version. So when updating this channel, it may replace the topic with a completely separate one instead of updating the existing one, which will be very unexpected.

Reply:
I already removed all the source IDs counter @kollivier and I use the topic name and Youtube id on it as you suggested. please review it again. 

I have a question? can I run this script directly on the cheffing server, I can't fully test this script in my local, Youtube already block my IP on API request.